Classes/Command/OpenWeatherMapCommand.php

```diff
@@ -0,0 +1,217 @@
+    public string $city = '';
+    public int $recordStoragePage = 0;
+
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly RequestFactory $requestFactory,
+        private readonly CacheService $cacheService,
+        private readonly ConnectionPool $connectionPool,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setHelp('Calls the api of openweathermap.org and saves response into database')
+            ->addArgument('name', InputArgument::REQUIRED, 'Name')
+            ->addArgument('city', InputArgument::REQUIRED, 'City name (e.g. Munich)')
+            ->addArgument('country', InputArgument::REQUIRED, 'Country Code (e.g. DE)')
+            ->addArgument('apiKey', InputArgument::REQUIRED, 'API-Key')
+            ->addArgument(
+                'clearCache',
+                InputArgument::OPTIONAL,
+                'Clear cache for pages (comma separated list with IDs)',
+            )
+            ->addArgument('recordStoragePage', InputArgument::OPTIONAL, 'Record storage page (optional)');
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $output->writeln('<info>Starting OpenWeatherMap data fetch...</info>');
+
+        try {
+            // Assign arguments
+            $this->name = ($input->getArgument('name')) ?? '';
+            $this->city = ($input->getArgument('recordStoragePage')) ?? '';
+            $this->country = ($input->getArgument('recordStoragePage')) ?? '';
+            $this->recordStoragePage = (int)$input->getArgument('recordStoragePage');
+
+            $this->removeOldRecordsFromDb();
+            $this->url = sprintf(
+                'https://api.openweathermap.org/data/2.5/weather?q=%s,%s&units=%s&APPID=%s',
+                urlencode($input->getArgument('city')),
+                urlencode($input->getArgument('country')),
+                'metric',
+                $input->getArgument('apiKey'),
+            );
+
+            // Log request details
+            $this->logger->info('Requesting data from OpenWeatherMap API', ['url' => $this->url]);
+
+            // Make API request
+            $response = $this->requestFactory->request($this->url);
+
+            if (!($this->checkResponseCode($response))) {
+                return Command::FAILURE;
+            }
+
+            $this->responseClass = json_decode((string)$response->getBody());
+            $this->logger->info(sprintf('Response class: %s', json_encode($this->responseClass)));
+
+            // Changing the data save to query builder
+            $this->saveCurrentWeatherInstanceForResponseClass($this->responseClass);
+
+            if (!empty($input->getArgument('clearCache'))) {
+                $cacheService = $this->cacheService;
+                $cacheService->clearPageCache(GeneralUtility::intExplode(',', $this->clearCache));
```

Comment:
What?! "ClearCache" sound like a bool value. "intExplode" on a bool value?

Ahh. It seems to clear the cache of a list of page UIDs. Please update that variable name